subsys/bluetooth/host/l2cap.c

```diff
@@ -1167,25 +1177,34 @@ static void le_ecred_conn_req(struct bt_l2cap *l2cap, uint8_t ident,
 	}
 
 response:
-	buf = l2cap_create_le_sig_pdu(buf, BT_L2CAP_ECRED_CONN_RSP, ident,
-				      sizeof(*rsp) + (sizeof(scid) * i));
-
-	rsp = net_buf_add(buf, sizeof(*rsp));
-	(void)memset(rsp, 0, sizeof(*rsp));
-
 	if (result == BT_L2CAP_LE_ERR_UNACCEPT_PARAMS ||
 	    result == BT_L2CAP_LE_ERR_PSM_NOT_SUPP ||
-	    result == BT_L2CAP_LE_ERR_AUTHENTICATION) {
-		memset(dcid, 0, sizeof(scid) * i);
+	    result == BT_L2CAP_LE_ERR_AUTHENTICATION ||
```

Comment:
This code path seems very confusing. Is there any error codes that requires special handling here?
It would seem to be that this should be `if (result != BT_L2CAP_LE_ERR_SUCCESS)`.

Reply:
I wondered the same, so if there's no special cases I think it should be as you suggested